src/Service/Modules/Reports/PurchaseOrdersReportModule.cs

```diff
@@ -11,23 +14,35 @@
     using Carter.Response;
 
     using Linn.Common.Facade;
+    using Linn.Common.Serialization;
     using Linn.Purchasing.Facade.Services;
     using Linn.Purchasing.Resources;
     using Linn.Purchasing.Service.Extensions;
     using Linn.Purchasing.Service.Models;
 
     using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Http.Extensions;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
+    using Microsoft.Extensions.FileProviders;
     using Microsoft.Extensions.Primitives;
 
     public class PurchaseOrdersReportModule : CarterModule
     {
         private readonly IPurchaseOrderReportFacadeService purchaseOrderReportFacadeService;
 
+        private readonly CsvStreamWriter csvStreamWriter;
+
+        private MemoryStream stream;
+
         public PurchaseOrdersReportModule(IPurchaseOrderReportFacadeService purchaseOrderReportFacadeService)
         {
             this.purchaseOrderReportFacadeService = purchaseOrderReportFacadeService;
             this.Get("/purchasing/reports/orders-by-supplier", this.GetApp);
             this.Get("/purchasing/reports/orders-by-supplier/report", this.GetOrdersBySupplierReport);
+            this.Get("/purchasing/reports/orders-by-supplier/export", this.GetOrdersBySupplierExport);
+            this.stream = new MemoryStream();
+            this.csvStreamWriter = new CsvStreamWriter(this.stream);
```

Comment:
Could this not come in via dependency injection?

Reply:
I mean the StreamWriter, not the Stream. I'd declare the stream at the point of use inside a using block as per above comment.